adapter/src/main/java/org/apache/calcite/prepare/KuduPrepareImpl.java

```diff
@@ -488,13 +489,12 @@ static void setFinalStatic(Field field, Object newValue) throws Exception {
   }
 
   // TODO figure out a better way to set the HintStrategyTable for sql queries
-  static {
+  public static void initializeHints() {
     try {
-      HintStrategyTable hintStrategyTable = HintStrategyTable.builder()
```

Comment:
Would we have to change existing Audit events queries to explicitly add hint ?

Reply:
Yes we can change the code that calls Relbuilder to add a hint when querying tables that need to use the KuduNestedJoin.

Reply:
1. Will the change be to add the `initializeHints` method around [RelBuilder.create](https://code.hq.twilio.com/data/KangarooJack/blob/master/data-engine-server/src/main/java/com/twilio/raas/query/service/QueryPlanner.java#L255) depending on the table name?
2. Will the hint be always set for Organization queries? Where do we set it for Organization queries?

Reply:
In KuduQuery since we do we don't need to call `initializeHints`  while using RelBuilder to build a query. We only need it if you execute a sql query and want to specify the hint. 
```
getCluster().setHintStrategies(KUDU_HINT_STRATEGY_TABLE);
```
The hint will be set in the query planner code right after we do the join
```
this.builder.join(JoinRelType.INNER, joinCondition);
        // force the use of KuduNestedJoin
        RelHint hint =
            RelHint.builder("USE_KUDU_NESTED_JOIN")
                .build();
```